llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -687,6 +691,15 @@ class TargetTransformInfoImplBase {
     return 1;
   }
 
+  InstructionCost getStridedMemoryOpCost(unsigned Opcode, Type *DataTy,
+                                         const Value *Ptr, bool VariableMask,
+                                         Align Alignment,
+                                         TTI::TargetCostKind CostKind,
+                                         const Instruction *I = nullptr) const {
+    return CostKind == TTI::TCK_RecipThroughput ? TTI::TCC_Expensive
```

Comment:
As I said in the review before you split, I would strongly prefer if this returned InstructionCost::getInvalid by default.  TCC_Basic is fine for codesize of a legal strided access, but all others should return invalid.